File: src/java/io/FileDescriptor.ts
export class FileDescriptor {
// Constructors
public constructor() {
return new Packages.java.io.FileDescriptor()
}
// Constructors
public constructor(int: number) {
return new Packages.java.io.FileDescriptor(int)
}
// Methods
private static dupFd(int: number): FileDescriptor {
return Packages.java.io.FileDescriptor.dupFd(int)
}
private static native isSocket(int: number): boolean {
return Packages.java.io.FileDescriptor.isSocket(int)
}
public getInt$(): number {
return this.getInt$()
}
public isSocket$(): boolean {
return this.isSocket$()
}
public setInt$(int: number): void {
this.setInt$(int)
}
public native sync(): void {
this.sync()
}
public valid(): boolean {
return this.valid()
}
// Fields
private descriptor: number
public static err: FileDescriptor
public static in: FileDescriptor
public static out: FileDescriptor
}

File: src/java/io/FileDescriptor.java
public final class java.io.FileDescriptor
{
    // Constructors
    public final java.io.FileDescriptor();
    public final java.io.FileDescriptor(int);
    // Methods
    private static java.io.FileDescriptor dupFd(int);
    private static native boolean isSocket(int);
    public final int getInt$();
    public boolean isSocket$();
    public final void setInt$(int);
    public native void sync();
    public boolean valid();
    // Fields
    private int descriptor;
    public static final java.io.FileDescriptor err;
    public static final java.io.FileDescriptor in;
    public static final java.io.FileDescriptor out;
}